builders/pallets-precompiles/precompiles/call-permit.md

```diff
@@ -0,0 +1,456 @@
+      to: to,
+      value: value,
+      data: data,
+      gaslimit: gaslimit,
+      nonce: nonce,
+      deadline: deadline,
+    };
+
+    const typedData = JSON.stringify({
+      types: {
+        EIP712Domain: [{
+            name: "name",
+            type: "string"
+          },
+          {
+            name: "version",
+            type: "string"
+          },
+          {
+            name: "chainId",
+            type: "uint256"
+          },
+          {
+            name: "verifyingContract",
+            type: "address"
+          },
+        ],
+        CallPermit: [{
+            name: "from",
+            type: "address"
+          },
+          {
+            name: "to",
+            type: "address"
+          },
+          {
+            name: "value",
+            type: "uint256"
+          },
+          {
+            name: "data",
+            type: "bytes"
+          },
+          {
+            name: "gaslimit",
+            type: "uint64"
+          },
+          {
+            name: "nonce",
+            type: "uint256"
+          },
+          {
+            name: "deadline",
+            type: "uint256"
+          },
+        ],
+      },
+      primaryType: "CallPermit",
+      domain: {
+        name: "Call Permit Precompile",
+        version: "1",
+        chainId: {{ networks.moonbase.chain_id }},
+        verifyingContract: "{{ networks.moonbase.precompiles.call_permit }}",
+      },
+      message: message,
+    });
+
+    return {
+      typedData,
+      message,
+    };
+  };
+
+  const method = "eth_signTypedData_v4";
+  const messageData = createPermitMessageData();
+  const params = [from, messageData.typedData];
+
+  web3.currentProvider.sendAsync({
+      method,
+      params,
+      from,
+    },
+    function(err, result) {
+      if (err) return console.dir(err);
+      if (result.error) {
+        alert(result.error.message);
+        return console.error("ERROR", result);
+      }
+      console.log("Signature:" + JSON.stringify(result.result));
+      console.log(ethers.utils.splitSignature(result.result))
+
+    }
+  );
+}
+
+main()
+```
+
+To run the code, click **Run** at the top of the page (or you can also use `control` and `s`). MetaMask should pop up and prompt you to connect an account. Make sure to choose the account you want to sign the message with. Then go ahead and sign the message.
+
+![Sign the message with MetaMask](/images/builders/pallets-precompiles/precompiles/call-permit/call-8.png)
+
+Once you've signed the message, go back to JSFiddle and if the console isn't already open, go ahead and open it to see the signature values include the `v`, `r`, and `s`, values. Copy these values as you'll need them when interacting with the call permit precompile in the following sections.
+
+![Signature values in the JSFiddle console](/images/builders/pallets-precompiles/precompiles/call-permit/call-9.png)
+
+### Use MetaMask's JS Signing Library {: #use-metamasks-signing-library }
```

Comment:
I would add a note here that this is only for example purposes and you should never share your private key (same disclaimers we've added in other parts)